src/app/reporting/Engine.cpp

```diff
@@ -290,21 +291,92 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
     return err;
 }
 
+CHIP_ERROR Engine::RemoveUnaccessableEventPaths(TLV::TLVWriter & aWriter, bool & aItemRemoved, ReadHandler * apReadHandler)
+{
+    CHIP_ERROR err                                                      = CHIP_NO_ERROR;
+    ObjectList<EventPathParams> * deletedEventPathList                  = nullptr;
+    ObjectList<EventPathParams> ** previousIteratedEventPathListNextPtr = &(apReadHandler->mpEventPathList);
+
+    aItemRemoved = false;
+    for (auto current = apReadHandler->mpEventPathList; current != nullptr;)
+    {
+        if (current->mValue.HasEventWildcard())
+        {
+            current = current->mpNext;
```

Comment:
Don't we need to update `previousIteratedEventPathListNextPtr` here?  Why did this not get caught by tests?